From a09cdc9f4b85995e83566a018049d9064579b22c Mon Sep 17 00:00:00 2001
From: Gao Pan <gaopan@freescale.com>
Date: Thu, 11 Jun 2015 11:33:13 +0800
Subject: [PATCH] MLK-11098: arm: imx: set imx6qp eim_slow to 132Mhz

eim_slow is 66MHz on imx6qp and 132MHz on imx6q. As a result,
imx6qp weim nor read performance is half of imx6q.

To eliminate the performance difference, imx6qp eim_slow is set
to 132Mhz.

Signed-off-by: Gao Pan <b54642@freescale.com>
---
 arch/arm/mach-imx/clk-imx6q.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-imx/clk-imx6q.c b/arch/arm/mach-imx/clk-imx6q.c
index e74332d..3348284 100644
--- a/arch/arm/mach-imx/clk-imx6q.c
+++ b/arch/arm/mach-imx/clk-imx6q.c
@@ -867,6 +867,9 @@ static void __init imx6q_clocks_init(struct device_node *ccm_node)
 	if (IS_ENABLED(CONFIG_PCI_IMX6))
 		imx_clk_set_parent(clk[IMX6QDL_CLK_LVDS1_SEL], clk[IMX6QDL_CLK_SATA_REF]);
 
+	/* set eim_slow to 132Mhz */
+	imx_clk_set_rate(clk[IMX6QDL_CLK_EIM_SLOW], 132000000);
+
 	/*
 	 * Enable clocks only after both parent and rate are all initialized
 	 * as needed
-- 
1.8.0

